core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -474,9 +495,9 @@ class DAGSchedulerSuite
 
   test("run trivial shuffle with fetch failure") {
     val shuffleMapRdd = new MyRDD(sc, 2, Nil)
-    val shuffleDep = new ShuffleDependency(shuffleMapRdd, null)
+    val shuffleDep = new ShuffleDependency(shuffleMapRdd, new HashPartitioner(2))
```

Comment:
These changes were needed because the new MapOutputTracker.getPreferredLocationsForShuffle figures out the number of partitions from the shuffle dependency's partitioner, so we could no longer pass in null as the partitioner.